SmartDeviceLinkTests/ProxySpecs/SDLSecondaryTransportManagerSpec.m

```diff
@@ -483,36 +488,62 @@ + (void)swapConnectionMethods {
         });
     });
 
-
     describe(@"In Configured state", ^{
         describe(@"if secondary transport is iAP", ^{
             beforeEach(^{
                 // in this case we assume the primary transport is TCP
                 testPrimaryProtocol = [[SDLProtocol alloc] init];
                 testPrimaryTransport = [[SDLTCPTransport alloc] init];
                 testPrimaryProtocol.transport = testPrimaryTransport;
+
                 dispatch_sync(testStateMachineQueue, ^{
                     [manager startWithPrimaryProtocol:testPrimaryProtocol];
                 });
 
                 manager.secondaryTransportType = SDLTransportSelectionIAP;
             });
 
-            it(@"should transition to Connecting state", ^{
-                // setToState cannot be used here, as the method will set the state after calling didEnterStateConfigured
-                dispatch_sync(testStateMachineQueue, ^{
-                    [manager.stateMachine transitionToState:SDLSecondaryTransportStateConfigured];
+            context(@"before the security manager is set by register app interface response", ^{
+                it(@"should transition to Connecting", ^{
+                    // setToState cannot be used here, as the method will set the state after calling didEnterStateConfigured
+                    dispatch_sync(testStateMachineQueue, ^{
+                        [manager.stateMachine transitionToState:SDLSecondaryTransportStateConfigured];
+                    });
+
+                    expect(manager.stateMachine.currentState).to(equal(SDLSecondaryTransportStateConnecting));
+                    expect(manager.secondaryProtocol.securityManager).to(beNil());
+                    expect(manager.isAppReady).to(equal(NO));
+
+                    OCMVerifyAll(testStreamingProtocolDelegate);
                 });
+            });
 
-                expect(manager.stateMachine.currentState).to(equal(SDLSecondaryTransportStateConnecting));
-                OCMVerifyAll(testStreamingProtocolDelegate);
+            context(@"after the security manager is set by register app interface response", ^{
+                it(@"should transition to Connecting state", ^{
+                    testPrimaryProtocol.securityManager = OCMClassMock([SDLFakeSecurityManager class]);
```

Comment:
This should be in a beforeEach in the context.

Reply:
Done